src/sentry/integrations/messaging.py

```diff
@@ -0,0 +1,215 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+from django.urls import re_path
+from django.urls.resolvers import URLPattern
+from django.views.generic import View
+
+from sentry import analytics
+from sentry.incidents.action_handlers import ActionHandler, DefaultActionHandler
+from sentry.incidents.models.alert_rule import ActionHandlerFactory, AlertRuleTriggerAction
+from sentry.incidents.models.incident import Incident, IncidentStatus
+from sentry.integrations.base import IntegrationProvider
+from sentry.models.notificationaction import ActionService, ActionTarget
+from sentry.models.project import Project
+from sentry.rules import rules
+from sentry.rules.actions import IntegrationEventAction
+
+
+@dataclass(frozen=True)
+class MessagingIdentityLinkViewSet:
+    """An integration's set of view classes for linking and unlinking identities."""
+
+    link_personal_identity: type[View]
+    unlink_personal_identity: type[View]
+
+    # TODO: Allow these to be optional? Require for all messaging integrations?
```

Comment:
Yes! For anything that we *want* to be a consistent feature across all messaging integrations, it would be good to represent it in the spec and to communicate whether it's optional. Filling in each unimplemented method can be self-contained tasks, and when they're all implemented we can switch it to be required.